bazel/README.md

```diff
@@ -71,13 +71,42 @@ tools/bazel-test-gdb //test/common/http:async_client_impl_test
 
 # Additional Envoy build and test options
 
+In general, there are 3 [compilation
+modes](https://bazel.build/versions/master/docs/bazel-user-manual.html#flag--compilation_mode)
+that Bazel supports:
+
+* `fastbuild`: `-O0 -DDEBUG`, aimed at developer speed (default).
+* `opt`: `-O2`, for production builds and performance benchmarking.
```

Comment:
I've added `--define debug_symbols=yes` support. I actually think it probably would be just as valid to have just suggested that the invoker provider `--copt=-ggdb3` as an extra `bazel build` arg, but this works too.